### ISSUE TITLE
Single partintro paragraph is parsed incorrectly

### ISSUE THREAD

**Issue by owenh000** [asciidoctor/asciidoctor#2860]
Asciidoctor uses the `partintro` style for paragraphs that appear immediately after a part title. The `partintro` style can be set manually or is set automatically (issue #768).

However, when the `partintro` style is used on a single paragraph, whether manually or automatically, the block context is set to `open` and the actual paragraph block is never processed as a paragraph node. Thus, for the HTML5 output, the paragraph text is not surrounded by `<p>`/`</p>` tags and the `p` CSS style is not applied as it should be (including the `font-size:1.0625rem` at [asciidoctor-default.css, line 382](https://github.com/asciidoctor/asciidoctor/blob/dd5c0e38a011d97295c37075b7ec7331aa7e21cd/data/stylesheets/asciidoctor-default.css#L382)).

Based on code and comments in [parser.rb](https://github.com/asciidoctor/asciidoctor/blob/622dfd08696791e74c61d7f3e0ff56d6ba40b772/lib/asciidoctor/parser.rb), it appears that the `partintro` style is intended to be handled with an open block regardless of whether it is a single paragraph or multiple. In both cases then, the paragraph(s) should be child blocks of the open block. Currently, if a single paragraph has the `partintro` style, the block context is simply changed to `open`. For the case of the HTML5 converter, this results in the block never being processed by the paragraph() method. Perhaps something is needed around [parser.rb, line 771](https://github.com/asciidoctor/asciidoctor/blob/622dfd08696791e74c61d7f3e0ff56d6ba40b772/lib/asciidoctor/parser.rb#L771)?

### Example 1: single paragraph, missing p tags, incorrect

Input:
```
= Test
:doctype: book

= Part 1

[partintro]
Child paragraph 1 of 1

== Chapter 1
```

Output excerpt:
```
<h1 id="_part_1" class="sect0">Part 1</h1>
<div class="openblock partintro">
<div class="content">
Child paragraph 1 of 1
</div>
</div>
```

### Example 2: two paragraphs, p tags present, correct

Input:
```
= Test
:doctype: book

= Part 1

Child paragraph 1 of 2

Child paragraph 2 of 2

== Chapter 1
```

Output excerpt:
```
<h1 id="_part_1" class="sect0">Part 1</h1>
<div class="openblock partintro">
<div class="content">
<div class="paragraph">
<p>Child paragraph 1 of 2</p>
</div>
<div class="paragraph">
<p>Child paragraph 2 of 2</p>
</div>
</div>
</div>
```

**Comment by owenh000** [comment]
The corresponding test at [sections_test.rb, line 3315](https://github.com/asciidoctor/asciidoctor/blob/622dfd08696791e74c61d7f3e0ff56d6ba40b772/test/sections_test.rb#L3315) should be extended as follows:

```diff
-    test 'should add partintro style to child paragraph of part' do
+    test 'should add partintro style to single child paragraph of part' do
       input = <<-EOS
 = Book
 :doctype: book
@@ -3328,6 +3328,8 @@ part intro
       partintro = doc.blocks.first.blocks.first
       assert_equal :open, partintro.context
       assert_equal 'partintro', partintro.style
+      assert_equal 1, partintro.blocks.size
+      assert_equal :paragraph, partintro.blocks[0].context
     end
```

**Comment by owenh000** [comment]
It is possible to work around this problem by using an explicit open block.

```
= Test
:doctype: book

= Part 1

[partintro]
--
Child paragraph 1 of 1
--

== Chapter 1
```

**Comment by owenh000** [comment]
Yes, in your example also the expected styling is missing. I agree that this should be changed, and that semantic HTML in #242 seems to be the best way to do that.

**Comment by hepabolu** [comment]
I came across this issue. For now I've added a workaround by adding the following as an invisible second paragraph:

```
ifndef::backend-epub3[]
+++&nbsp;+++
endif::[]
```

It does create extra whitespace, but at least the first paragraph is rendered correctly
